x-pack/plugins/security_solution/server/endpoint/routes/policy/service.ts

```diff
@@ -57,3 +60,70 @@ export async function getPolicyResponseByAgentId(
     policy_response: response.hits.hits[0]._source,
   };
 }
+
+export async function getAgentPolicySummary(
+  endpointAppContext: EndpointAppContext,
+  soClient: SavedObjectsClientContract,
+  packageName: string,
+  policyId?: string,
+  pageSize: number = 1000
+): Promise<JsonObject> {
+  const agentVersionMapToJson = (versionMap: Map<string, number>): JsonObject => {
```

Comment:
Function seems to be pure. consider making this function a module level static instead fo re-defining it every time this method is called